docs/r3/example_design_for_atomic_swap.md

```diff
@@ -0,0 +1,212 @@
+* In cases where the finality status of a block can be determined directly by analysing the set of (Proof of Authority) identities that have signed it, we refer to these actors (who are members of the EVM netweork) as _Validators_.
+* In cases where the finality status of a block cannot be determined directly (as is common in Proof of Work and some Proof of Stake networks), we may rely on the opinion or one or more other observers on the Corda network, which we refer to as _Remote Finality Guarantors_, or just "Guarantors" when the context is clear.
+  * In such situations this service would be provided by extending the role of the Oracle so that it not only locates the requested block but verifies that it has reached finality. Assuming so, the Oracle signs the n-Tuple (blockchain id, block header, finality parameter) and returns the signature to the requesting party (blockchain id as per EIP-3220).
+  * The architecture paper elaborates on potential models for the implementation of [remote finality guarantors](remote_finality_guarantors.md).
+
+## Detailed Atomic Swap Protocol Flow
+The atomic swap protocol involves a detailed series of steps for asset exchange. They are depicted graphically in the sequence diagram below, and then elaborated on in the text
+
+![image](img/atomic-dvp-sequence-diagram.jpeg)
+_High level overview of the assumed network_
+
+### Agreement
+The atomic swap begins with an agreement between the two parties, where they have already agreed on:
+
+* The expected recipient of the EVM asset - This is an Externally Owned Address (EOA) maintained by the asset seller.
+* The EVM contract that facilitates the swap (swap contract for EVM asset commit and transfer)
+* Either:
+  * In the case of Proof of Authority, the identities of the _m_ EVM Validators, and a number _n <= m_ that must have signed a block for it to be considered final; _or_
+  * When block finality is harder to determine, the identities of the _m_ Remote Finality Guarantors on the Corda network, with a number _n <= m_ that must attest the finality of a block for it to be considered final on the Corda network. 
+    * An Oracle identity is in the form of a Corda Public Key, and an EVM address so that the Oracle is able to produce signatures that can be verified both on the Corda network and the EVM network as required.
+
+### Draft Transaction Creation and Verification
+
+Once the swap agreement is defined, the asset seller starts by building a Draft Corda Transaction that is then presented to the other party's Corda node for inspection. To proceed, asset purchaser must approve the Draft Transaction. Approval implies satisfaction that:
+
+* The transaction, once committed, would transfer the agreed asset from the asset seller's Corda node to the asset purchaser's Corda node
```

Comment:
I think...the transaction is the creation of an encumbered asset that on unlock would transfer the agreed asset from ...

Reply:
i've resequenced and reworded to make it clearer